.github/workflows/build-cibw.yml

```diff
@@ -0,0 +1,175 @@
+            platform_id: manylinux_x86_64
+            manylinux_image: manylinux2014
+          - os: ubuntu-latest
+            python_version: "3.11"
+            cibw_python_version: 311
+            platform_id: manylinux_x86_64
+            manylinux_image: manylinux2014
+          - os: ubuntu-latest
+            python_version: "3.12"
+            cibw_python_version: 312
+            platform_id: manylinux_x86_64
+            manylinux_image: manylinux2014
+          - os: ubuntu-latest
+            python_version: "3.13"
+            cibw_python_version: 313
+            platform_id: manylinux_x86_64
+            manylinux_image: manylinux2014
+
+          # Linux aarch64
+          - os: ubuntu-24.04-arm
+            python_version: "3.10"
+            cibw_python_version: 310
+            platform_id: manylinux_aarch64
+            manylinux_image: manylinux2014
+          - os: ubuntu-24.04-arm
+            python_version: "3.11"
+            cibw_python_version: 311
+            platform_id: manylinux_aarch64
+            manylinux_image: manylinux2014
+          - os: ubuntu-24.04-arm
+            python_version: "3.12"
+            cibw_python_version: 312
+            platform_id: manylinux_aarch64
+            manylinux_image: manylinux2014
+          - os: ubuntu-24.04-arm
+            python_version: "3.13"
+            cibw_python_version: 313
+            platform_id: manylinux_aarch64
+            manylinux_image: manylinux2014
+
+          # MacOS x86_64
+          # - os: macos-13
+          #   python_version: "3.10"
+          #   cibw_python_version: 310
+          #   platform_id: macosx_x86_64
+          # - os: macos-13
+          #   python_version: "3.11"
+          #   cibw_python_version: 311
+          #   platform_id: macosx_x86_64
+          # - os: macos-13
+          #   python_version: "3.12"
+          #   cibw_python_version: 312
+          #   platform_id: macosx_x86_64
+          # - os: macos-13
+          #   python_version: "3.13"
+          #   cibw_python_version: 313
+          #   platform_id: macosx_x86_64
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python_version }}
+
+      # Set the DEVELOP flag and the TIMESTAMP environment variables. This is used in the
+      # top-level CMakeLists.txt to generate the GTSAM_VERSION_STRING.
+      - name: Set Develop Flag
+        run: |
+          echo "DEVELOP=1" >> $GITHUB_ENV
+          echo "TIMESTAMP=${{ needs.get_system_time.outputs.timestamp }}" >> $GITHUB_ENV
+
+      - name: Install Dependencies
+        run: |
+          python3 -m pip install -r python/dev_requirements.txt
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            sudo apt-get install -y wget libicu-dev python3-pip python3-setuptools libboost-all-dev ninja-build
```

Comment:
This is fishy. You're installing `pip` after running `pip install`?
Probably should be a step before `Install Dependencies`.